.github/workflows/test.yml

```diff
@@ -63,51 +87,38 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/setup
-      - uses: actions/download-artifact@v2
+      - uses: actions/download-artifact@v3
         with:
           name: build
           path: build
       - run: yarn test:size
 
 
-  cypress-build:
+  cypress-test-matrix:
```

Comment:
cypress-build was basically a dupe step?

Reply:
cypress-build was building the cypress binary and caching it, and cypress-test was using that cached binary.
however, cypress-build added setup time to the total e2e testing time. moving the build to the cypress-test step means it will run slower with a cache miss but faster overall because it can skip a whole job's setup.